cl/risc0_proofs/nullifier/src/main.rs

```diff
@@ -0,0 +1,28 @@
+/// Nullifier Proof
+///
+/// Our goal: prove the nullifier nf was derived from a note that had previously been committed to.
+///
+/// More formally, nullifier statement says:
+/// for public input `nf` (nullifier) and `root_cm` (root of merkle tree over commitment set).
+/// the prover has knowledge of `output = (note, nf_pk, nonce)`, `nf` and `path` s.t. that the following constraints hold
+/// 0. nf_pk = hash(nf_sk)
+/// 1. nf = hash(nonce||nf_sk)
+/// 2. note_cm = output_commitment(output)
+/// 3. verify_merkle_path(note_cm, root, path)
+use cl::merkle;
+use cl::nullifier::Nullifier;
+use proof_statements::nullifier::{NullifierPrivate, NullifierPublic};
+use risc0_zkvm::guest::env;
+
+fn main() {
+    let secret: NullifierPrivate = env::read();
+    assert_eq!(secret.output.nf_pk, secret.nf_sk.commit());
```

Comment:
why take `nf_pk` as input if we recalculate it anyway to check consistency?

Reply:
You're right, we don't need this assert

Reply:
(The public output is the nullifier which is different from nf_pk)